src/lib.rs

```diff
@@ -732,6 +736,13 @@ impl Queue {
         }
     }
 
+    /// Set whether the recv function blocks or not
+    pub fn set_non_blocking(&mut self, non_blocking: bool) {
+        if non_blocking {
+            self.recv_flag = MSG_DONTWAIT;
```

Comment:
Does this have to be a per-call flag? Wouldn O_NONBLOCK via fcntl work?

Reply:
Also this looks wrong since MSG_TRUNC is left out, and you can't re-enable blocking with `set_nonblocking(false)`.

Reply:
Added the re-enable blocking code. Regarding the O_NONBLOCK via fcntl, it sure looks like it'd work. But I have neither the skills nor the time to add such a solution atm. I basically just copied what I found here and thought it'd be a good addition: https://github.com/kti/python-netfilterqueue/pull/10/commits/b14fa6484d3542e2530c0688daaf783b5a03227b#diff-609e4e568077f5fe828c1a78abdb3be64ef0810e90aa06b4195c9eba40f9541a

Reply:
The nonblocking flag needs to be `MSG_DONTWAIT | MSG_TRUNC`.

Reply:
I'm not quite sure what you mean. Are you saying something like changing the `non_blocking` argument to accept `MSG_DONTWAIT` or `MSG_TRUNC` instead of `bool`?